### ISSUE TITLE
More tightly couple actions and subsytems

### ISSUE THREAD

**Issue by sjcbulldog** [errorcodexero/newarch#81]
The first season with the new architecture has lead to some improvements we can make.  This is one of these improvements.

Currently actions and subsystems are loosely coupled.  This puts a heavy burden on the programmer to understand how the hierarchical subsystems interact and when specific actions can be run in parallel with other actions.  Coupling actions and subsystems will provide immediate feedback to the programmer when a sequence that is illegal is requested.  There are several things that need to happen here.
- Remove the DispatchAction class
- Provide an optional subsystem for all actions.
- Define default actions for subsystems
- Define the policy for actions preempting other actions
- Possibly create an ActionManager for dealing with deferred actions

**Comment by sjcbulldog** [comment]
I am marking this as complete as we revisited the way actions work and made the desired changes.  It is not exactly what is documented above, but it is new and improved and documented on the WiKi.
